utils/rjson.cc

```diff
@@ -242,23 +242,15 @@ class output_stream_buffer {
 public:
     output_stream_buffer(seastar::output_stream<char>& os) : _os(os) {}
     using Ch = char; // Used by rjson internally
-    future<> f = make_ready_future<>();
 
     void Flush() {
-        if (f.failed()) {
-            f.get();
-        }
         if (_pos == 0) {
             return;
         }
         if (_pos < _buf_size) {
             _buf.trim(_pos);  // Last flush may be shorter
         }
-        // Either we call futures right away (if they are ready) or we start growing continuations
-        // chain as we don't have the ability to wait here because Flush() signature is set by rjson.
-        f = f.then([this, b = std::move(_buf)] () mutable {
-            return send(std::move(b));
-        });
+        send(std::move(_buf)).get();
         _pos = 0;
         _buf = temporary_buffer<char>(_buf_size);
```

Comment:
with added .get() we can probably reuse the buffer

Reply:
We can use the buffer in _os. There's even a comment:

```c++
        // Note: Should consider writing directly to the buffer in output_stream
        // instead of double buffering. But output_stream for a single char has higher
        // overhead than the above check + once we hit a non-completed future, we'd have
        // to revert to this method anyway...
```

It's possible that now the overhead of writing directly the same.

I'll defer it to a follow-up though, it changes too many things.